File: components/credentials/src/main/java/org/cloudfoundry/credhub/audit/entities/InterpolateCredentials.java
package org.cloudfoundry.credhub.audit.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import org.cloudfoundry.credhub.audit.OperationDeviceAction;
import org.cloudfoundry.credhub.audit.RequestDetails;

@JsonIgnoreProperties(ignoreUnknown = true)
public class InterpolateCredentials implements RequestDetails {

  @Override
  public OperationDeviceAction operation() {
    return OperationDeviceAction.INTERPOLATE;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/audit/entities/InterpolateCredentials.kt
package org.cloudfoundry.credhub.audit.entities

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import org.cloudfoundry.credhub.audit.OperationDeviceAction
import org.cloudfoundry.credhub.audit.RequestDetails

@JsonIgnoreProperties(ignoreUnknown = true)
class InterpolateCredentials : RequestDetails {
    override fun operation(): OperationDeviceAction = OperationDeviceAction.INTERPOLATE
}
